ibm/service/power/data_source_ibm_pi_virtual_serial_number.go

```diff
@@ -0,0 +1,73 @@
+// Copyright IBM Corp. 2024 All Rights Reserved.
+// Licensed under the Mozilla Public License v2.0
+
+package power
+
+import (
+	"context"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+// Datasource to get a virtual serial number in a power instance
+func DataSourceIBMPIVirtualSerialNumber() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceIBMPIVirtualSerialNumberRead,
+		Schema: map[string]*schema.Schema{
+			// Arguments
+			Arg_CloudInstanceID: {
+				Description:  "The GUID of the service instance associated with an account.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+			Arg_Serial: {
+				Description:  "Virtual serial number.",
+				Required:     true,
+				Type:         schema.TypeString,
+				ValidateFunc: validation.NoZeroValues,
+			},
+
+			// Attributes
+			Attr_Description: {
+				Computed:    true,
+				Description: "Description of virtual serial number.",
+				Type:        schema.TypeString,
+			},
+			Attr_PVMInstanceID: {
```

Comment:
```suggestion
			Attr_PVMInstanceID: {
```

extremely minor nitpick, but we also have an attribute called Attr_InstanceID we use in other places that `instance_id` instead of `pvm_instance_id`